api/agent.go

```diff
@@ -888,3 +893,23 @@ func (a *Agent) updateToken(target, token string, q *WriteOptions) (*WriteMeta,
 	wm := &WriteMeta{RequestTime: rtt}
 	return wm, nil
 }
+
+// BootstrapGossipKey is used to deliver the gossip encryption key to the local
+// agent which has encrypt_key_from_api and disable_keyring_file both true. This
+// allows externally managing the key so it's never on disk, but requires that
+// external manager can deliver the correct key which is the current primary key
+// used in the cluster. Key is expected to be exactly 16 or 32 bytes encoded in
```

Comment:
32?

Reply:
Actually technically it can be 24 too.

https://github.com/hashicorp/memberlist/blob/2072f3a3ff4b7b3d830be77678d5d4b978362bc4/keyring.go#L61-L70